### PR TITLE
ESP32S2: Support for RotaryIO

### PR DESCRIPTION
**Pull request by microDev1**

### CODE REVIEW

File: ports/esp32s2/common-hal/rotaryio/IncrementalEncoder.c [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Scott Shawcroft for Adafruit Industries
```
Comment (tannewt) [inline ports/esp32s2/common-hal/rotaryio/IncrementalEncoder.c:6]: Update this for yourself.

**Review by tannewt — COMMENTED** [review]
This looks really good! Were you able to test it or do you need me to?

**Comment by microDev1** [comment]
@tannewt Thanks for the review. I tested this with my [microS2](https://circuitpython.org/board/microdev_micro_s2/) & two different types of encoder.

**Review by tannewt — CHANGES_REQUESTED** [review]
Don't forget to change the copyright. Let's wait for #3615 to be merged and then update this PR for the pcnt sharing code. Thanks!

**Comment by hierophect** [comment]
Seconding Scott here - if you can get everything in CountIO moved over to a `peripherals/pcnt.c` file, you can migrate all your init and reset functions over to it, and the two modules will be able to co-exist without accidentally trying to reserve each other's PCNT units. Check out rmt.c for an example - the RMT is used by PulseIn, PulseOut, *and* Neopixel_write, so they all share the same reservation system, init/deinit, and reset_all functions.

**Review by tannewt — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**